File: python/leetcode/minimum-time-difference.py
class Solution:
    def findMinDifference(self, timePoints):
        """
        :type timePoints: List[str]
        :rtype: int
        """
        minutes = []
        for t in timePoints:
            s = t.split(':')
            minutes.append((int(s[0]) * 60) + int(s[1]))
        full_cycle = 60 * 24
        min_dist = full_cycle
        for i in range(len(minutes)):
            for j in range(len(minutes)):
                if j != i:
                    dist = abs(minutes[i] - minutes[j])
                    dist = min([(minutes[i] + full_cycle) - minutes[j], dist])
                    dist = min([(minutes[j] + full_cycle) - minutes[i], dist])
                    if dist < min_dist:
                        min_dist = dist
        return min_dist

File: java/leetcode/minimum-time-difference.java
class Solution {
    public int findMinDifference(List<String> timePoints) {
        int[] minutes = new int[timePoints.size()];
        String[] s = new String[2]; // For split time strings
        for (int i = 0; i < timePoints.size(); i++) {
            s = timePoints.get(i).split(":");
            minutes[i] = (Integer.parseInt(s[0]) * 60) + Integer.parseInt(s[1]);
        }
        int fullCycle = 60 * 24;
        int minDist = fullCycle;
        int dist = 0;
        for (int k = 0; k < minutes.length; k++) {
            for (int j = 0; j < minutes.length; j++) {
                if (k != j) {
                    dist = Math.abs(minutes[k] - minutes[j]);
                    dist = Math.min(Math.abs(minutes[k] + fullCycle) - minutes[j], dist);
                    dist = Math.min(Math.abs(minutes[j] + fullCycle) - minutes[k], dist);
                    if (dist < minDist) { minDist = dist; }
                }
            }
        }
        return minDist;
    }
}
